### PR TITLE
feat(provider/kubernetes): V2 cluster provider

### PR DESCRIPTION
**Pull request by lwander**
@andrewbackes fyi

### CODE REVIEW

**Comment by lwander** [comment]
spinnaker/spinnaker#1902
spinnaker/spinnaker#1969

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/provider/KubernetesClusterProvider.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  @Override
+  public KubernetesCluster getCluster(String application, String account, String name, boolean includeDetails) {
+    return translateCluster(cacheUtils.getSingleEntry(CLUSTER.toString(), Keys.cluster(account, name)));
+  }
+
+  @Override
+  public KubernetesServerGroup getServerGroup(String account, String region, String name) {
+    // TODO(lwander)
+    return null;
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/provider/KubernetesClusterProvider.java]: Intentional, right? Just making sure.

**Comment by jtk54** [comment]
After tests pass.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/provider/KubernetesClusterProvider.java (hunk 1 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/provider/KubernetesClusterProvider.java]: Yup

**Pull request merged**